src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -440,7 +481,11 @@ public void DoWriteIfNeeded()
                 }, this);
 
                 Self._head = _lockedEnd.Block;
-                Self._head.Start = _lockedEnd.Index;
```

Comment:
Discovered race here; don't know if was with these changes or is in current build.